src/golang/lib/models/shared/metadata.go

```diff
@@ -0,0 +1,10 @@
+package shared
+
+type Metadata struct {
+	Schema []map[string]string // Table Schema from Pandas
+	// Metrics from the system regarding the op used to create the artifact result.
+	// A key/value pair of [metricname]metricvalue e.g. SystemMetric["runtime"] -> "3.65"
+	SystemMetrics     map[string]string `json:"system_metadata,omitempty"`
+	SerializationType SerializationType `json:"serialization_type,omitempty"`
+	ArtifactType      Type     `json:"artifact_type,omitempty"`
```

Comment:
Not sure where to put the metadata struct because it depends on the artifact's type struct (which I've put in shared).
- If I put metadata struct in utils, there is a circular import (importing from shared which is imported by utils)
- If I put metadata in shared, I need to import metadata in utils which is another import cycle.

Reply:
Does leaving `ArtifactType` and `Metadata` in shared work? Ideally `utils` shouldn't be importing `shared`, since it only defines primitive types and the serialization logic for all JSONB types.

Reply:
I can try doing that. Metadata has a NullMetadata with its own Value() and Scan() which is why I wanted to put that in primitives.